decimal.go

```diff
@@ -133,18 +137,28 @@ func NewFromString(value string) (Decimal, error) {
 		expInt := -len(decimalPart)
 		exp += int64(expInt)
 	} else {
-		return Decimal{}, fmt.Errorf("can't convert %s to decimal: too many .s", value)
+		return Decimal{}, &ErrorInvalidFormat{
+			reason: fmt.Sprintf(
+				"can't convert %s to decimal: too many .s",
+				value),
+		}
 	}
 
 	dValue := new(big.Int)
 	_, ok := dValue.SetString(intString, 10)
 	if !ok {
-		return Decimal{}, fmt.Errorf("can't convert %s to decimal", value)
+		return Decimal{}, &ErrorInvalidFormat{
+			reason: fmt.Sprintf("can't convert %s to decimal", value),
+		}
 	}
 
 	if exp < math.MinInt32 || exp > math.MaxInt32 {
 		// NOTE(vadim): I doubt a string could realistically be this long
-		return Decimal{}, fmt.Errorf("can't convert %s to decimal: fractional part too long", originalInput)
+		return Decimal{}, &ErrorInvalidFormat{
```

Comment:
This should be `ErrorFractionLimit`, or `ErrorExponentLimit` if you decide to change the name.

Reply:
Done